Challenge-4/Bot_1/src/agent.spec.ts

```diff
@@ -0,0 +1,284 @@
+        New_value: "200",
+      },
+      LIQUIDATE_CF: {
+        Old_value: "100",
+        New_value: "200",
+      },
+    };
+
+    const metadata = {
+      "Collateral Asset - Collateral-1":
+        "0x0000000000000000000000000000000000001247",
+      "Collateral Asset - Collateral-2":
+        "0x0000000000000000000000000000000000012567",
+      "Collateral Asset - Collateral-3":
+        "0x0000000000000000000000000000000000001289",
+      "Collateral Asset - Collateral-4":
+        "0x0000000000000000000000000000000000001290",
+    };
+
+    const collateralAddresses = [
+      createAddress("0x1247"),
+      createAddress("0x12567"),
+      createAddress("0x1289"),
+      createAddress("0x1290"),
+    ];
+    setupMockProvider(collateralAddresses);
+
+    const nonGovEvents = [
+      "event Transfer(address indexed from, address indexed to, uint256 value)",
+      "event Approval(address indexed owner, address indexed spender, uint256 value)",
+      "event Deposit(address indexed user, address reciepient, uint256 amount)",
+    ];
+
+    nonGovEvents.forEach((eventSignature) => {
+      txEvent.addEventLog(eventSignature, createAddress("0xABCDEF"), [
+        createAddress("0x12345"),
+        createAddress("0x67890"),
+        1000,
+      ]);
+    });
+
+    txEvent.addEventLog(
+      "event SetGovernor(address indexed cometProxy, address indexed oldGovernor, address indexed newGovernor)",
+      CONFIGURATOR_PROXY,
+      [...mockArgs],
+    );
+    txEvent.addEventLog(
+      "event SetBorrowKink(address indexed cometProxy,uint64 oldKink, uint64 newKink)",
+      CONFIGURATOR_PROXY,
+      [mockArgs[1], 100, 200],
+    );
+    txEvent.addEventLog(
+      "event SetSupplyKink(address indexed cometProxy,uint64 oldKink, uint64 newKink)",
+      CONFIGURATOR_PROXY,
+      [mockArgs[1], 100, 250],
+    );
+    txEvent.addEventLog(
+      "event UpdateAssetBorrowCollateralFactor(address indexed cometProxy, address indexed asset, uint64 oldBorrowCF, uint64 newBorrowCF)",
+      CONFIGURATOR_PROXY,
+      [mockArgs[1], createAddress("0x123"), 100, 200],
+    );
+    txEvent.addEventLog(
+      "event UpdateAssetLiquidateCollateralFactor(address indexed cometProxy, address indexed asset, uint64 oldLiquidateCF, uint64 newLiquidateCF)",
+      CONFIGURATOR_PROXY,
+      [mockArgs[1], createAddress("0x123"), 100, 200],
+    );
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toEqual([
+      Finding.fromObject({
+        name: `Change of asset values due to governance proposal`,
+        description: `The asset ${mockAssetTokenAddress} has been modified by a governance proposal`,
+        alertId: "GOV-1",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: txEvent.network.toString(),
+        metadata: {
+          ...metadata1,
+          ...metadata,
+        },
+      }),
+    ]);
+    mockProvider.clear();
+    expect(findings.length).toStrictEqual(1);
```

Comment:
Again, not sure about this.

Reply:
To check if lru-cache implementation works correctly